libtock/tock.c

```diff
@@ -102,6 +103,24 @@ void yield_for(bool *cond) {
   }
 }
 
+int quick_yield(void* base) {
+  if (task_cur != task_last) {
+    tock_task_t task = task_queue[task_cur];
+    task_cur = (task_cur + 1) % TASK_QUEUE_SIZE;
+    task.cb(task.arg0, task.arg1, task.arg2, task.ud);
+    return 1;
+  } else {
+    uint32_t tasks = ros_get_pending_tasks(base);
+
+    if (tasks > 0) {
+      // Waiting tasks, call yield
+      yield();
+    }
+
+    return tasks;
+  }
+}
+
```

Comment:
I think there's some weird intermingling between the `tock` and `read_only_state` module here.

Can this possibly be decoupled a bit, i.e. maybe a `tock_execute_next_task()` method which executes the top half `if (task_cur != task_last)` case; that can be part of the `tock` module generally, then the `read_only_state_quick_yield(base)` can call that method and do the second half?